src/shared/components/cancerSummary/CancerSummary.tsx

```diff
@@ -0,0 +1,448 @@
+                    return {
+                        label: cancerTypeS,
+                        sortBy: yAxis,
+                        symbol: yAxis === "abs-count" ? '' : "%",
+                        sortCount: yAxis === "abs-count" ? altTotalCount : altTotalPercent,
+                        data: _.map(cancerAlterations, (count:number, altType: string) => {
+                            const percent = count / totalCases * 100;
+                            const total = yAxis === "abs-count" ? count : percent;
+                            return {
+                                label: altType,
+                                totalCases,
+                                altTotalPercent,
+                                total,
+                                count,
+                                percent,
+                                backgroundColor: showGenomicAlt  ? this.getColors(altType) : '#aaaaaa'
+                            };
+                        })
+                    };
+                }
+            }),
+            undefined, null
+        );
+    }
+
+    private handleYAxisChange() {
+        this.yAxis = this.inputYAxisEl.value;
+        this.resetSliders();
+    }
+
+    private handleXAxisChange() {
+        this.xAxis = this.inputXAxisEl.value;
+        this.resetSliders();
+    }
+
+    private resetSliders() {
+        this.altCasesValue = 0;
+        this.totalCasesValue = 0;
+    }
+
+    private handleGenomicCheckboxChange() {
+        this.showGenomicAlt = !this.showGenomicAlt;
+    }
+
+    private handleAltSliderChange(value:number) {
+        this.altCasesValue = value;
+    }
+
+    private handleTotalSliderChange(value:number) {
+        this.totalCasesValue = value;
+    }
+
+    private toggleShowControls() {
+        this.showControls = !this.showControls;
+    }
+
+    @computed private get cancerTypes() {
+        return [{label: 'All', value: 'all'},
+                ...Object.keys(this.props.data).map(point => (
+                        {label: point, value: point}
+                    ))
+                ];
+    }
+
+    private getColors(color:string) {
+        const colors:{[id:string]:string} = {
+            mutated:"#36a2eb",
+            amplified:"#ff6384",
+            deleted:"#ffce56",
+            multiple:"#cc65fe",
+        };
+        return colors[color];
+    }
+
+    private toRoundedNumber(decimal: string | number) {
+        return Math.round(Number(decimal) * 100) / 100;
+    }
+
+    @computed private get altCasesMax() {
+        return Math.max(...this.datasets.map(data => data!.sortCount));
+    }
+
+    @computed private get totalCasesMax() {
+        const value = this.multiSelectValue;
+        return Math.max(..._.map(
+            _.filter(this.props.data, (o, label) => (
+                !!_.intersection(["all", "", label.toLowerCase()], value.split(",").map(val => val.toLowerCase())).length
```

Comment:
since this is used above, maybe make a computed getter selectedValues or something, which handles split and lowercasing of values.